src/assets/wise5/authoringTool/constraint/constraint-authoring/constraint-authoring.component.ts

```diff
@@ -0,0 +1,237 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { MultipleChoiceContent } from '../../../components/multipleChoice/MultipleChoiceContent';
+import { TeacherDataService } from '../../../services/teacherDataService';
+import { TeacherProjectService } from '../../../services/teacherProjectService';
+
+@Component({
+  selector: 'constraint-authoring',
+  templateUrl: './constraint-authoring.component.html',
+  styleUrls: ['./constraint-authoring.component.scss']
+})
+export class ConstraintAuthoringComponent implements OnInit {
+  @Input() constraint: any;
+  constraintActions = [
+    { value: '', text: $localize`Please Choose an Action` },
+    {
+      value: 'makeAllNodesAfterThisNotVisitable',
+      text: $localize`Make all nodes after this not visitable`
+    },
+    {
+      value: 'makeAllNodesAfterThisNotVisible',
+      text: $localize`Make all nodes after this not visible`
+    },
+    { value: 'makeAllOtherNodesNotVisitable', text: $localize`Make all other nodes not visitable` },
+    { value: 'makeAllOtherNodesNotVisible', text: $localize`Make all other nodes not visible` },
+    { value: 'makeThisNodeNotVisitable', text: $localize`Make this node not visitable` },
+    { value: 'makeThisNodeNotVisible', text: $localize`Make this node not visible` }
+  ];
+  node: any;
+  nodeIds: string[];
+  removalConditionals = [
+    { value: 'all', text: $localize`All` },
+    { value: 'any', text: $localize`Any` }
+  ];
+  removalCriteria = [
+    { value: '', text: $localize`Please Choose a Removal Criteria` },
```

Comment:
Add params for consistency? Then, we won't need a null-check in removalCriteriaNameChanged()
```suggestion
    { value: '', text: $localize`Please Choose a Removal Criteria`, params: [] },
```